index.js

```diff
@@ -270,12 +276,22 @@ function getContentType(limit, skip, contentSettings, itemsPulled) {
                 ) {
                     fileContent += `---\n`;
                 }
+                
 
                 // if set add the main content below the front matter
                 if (item.fields[contentSettings.mainContent]) {
-                    fileContent += `${
-                        item.fields[contentSettings.mainContent]
-                    }`;
+                    content = item.fields[contentSettings.mainContent]
+                    if (typeof content === "object") {
+                        fileContent += richTextToMarkdown(content)
+                    } else {
+                        fileContent += content;
+                    }
+                }
+
+                // set the file name to fileNameField, if present
+                fileName = item.sys.id;
+                if (item.fields[contentSettings.fileNameField]) {
+                    fileName = item.fields[contentSettings.fileNameField]
```

Comment:
later on we'll want to have a type checker to makes sure that ```item.fields[contentSettings.fileNameField]``` isn't an object or something.

Also not an immediate need, but down the line it would also be worth it to "slugify" the filename. For example if someone uses the Title field as a fileNameField and their title is "What are the 5 Best Static Site Generators?" it get's turned into something like this ```what-are-the-5-best-static-site-generators.md```